docs/core/howto/defer-intro.rst

```diff
@@ -105,12 +105,113 @@ Unless we hack the interpreter somehow, we would need to build this with the Pyt
 
 Perhaps we want something that looks a little like this::
 
-    placeholder = pod_bay_doors.open()
-    placeholder.when_done(pod.launch)
+    async with pod_bay_doors.open():
+        await pod.launch()
```

Comment:
Maybe the example is easier to understand without the `with` statement:
```py
        await pod_bay_doors.open()
        pod.launch()
```
The `with` block might be better if this were actual code, since it could make sure the doors stay open until the pod has launched and close afterwards. But we're just illustrating a concept, not writing code for a spaceship ;)

Reply:
I think it's important to show all the features up front as early as possible in the document. But I'm happy to rollback on this change to get it landed

Reply:
It's not a big issue for me; if no-one else has a clear preference you can go with your solution.